specifications/Disruptor/Disruptor_SPMC.tla

```diff
@@ -0,0 +1,164 @@
+----------------------------- MODULE Disruptor_SPMC ------------------------
+(***************************************************************************)
+(* Models a Single Producer, Multi Consumer Disruptor (SPMC).              *)
+(*                                                                         *)
+(* The producer publishes the sequence number as value into the ring       *)
+(* buffer and the model verifies that all consumers read all published     *)
+(* values.                                                                 *)
+(*                                                                         *)
+(* The model also verifies that no data races occur between the producer   *)
+(* and consumers and that all consumers eventually read all published      *)
+(* values.                                                                 *)
+(*                                                                         *)
+(* To see a data race, try and run the model with two producers.           *)
+(***************************************************************************)
+
+EXTENDS Integers, FiniteSets, Sequences
+
+CONSTANTS
+  MaxPublished, (* Max number of published events. Bounds the model. *)
+  Writers,      (* Writer/producer thread ids.                       *)
+  Readers,      (* Reader/consumer thread ids.                       *)
+  Size,         (* Ringbuffer size.                                  *)
+  NULL
+
+ASSUME Writers /= {}
+ASSUME Readers /= {}
+ASSUME Size \in Nat \ {0}
+
+VARIABLES
+  ringbuffer,
+  published,    (* Write cursor. One for the producer.               *)
+  read,         (* Read cursors. One per consumer.                   *)
+  consumed,     (* Sequence of all read events by the Readers.       *)
```

Comment:
`consumed` is a history variable. It might be helpful to add a comment explaining its purpose to make it clearer for readers.

Reply:
Thanks, I'll add more comments explaining that. It's a new term to me but it makes total sense and it's nice to point out in the spec that's it's not part of the system being modeled but rather an auxiliary variable used for verification.